api/utils.go

```diff
@@ -58,29 +55,31 @@ func resolveBlockTag(
 }
 
 func resolveBlockNumber(
-	number rpc.BlockNumber,
+	blockNumber rpc.BlockNumber,
 	blocksDB storage.BlockIndexer,
 ) (uint64, error) {
-	height := number.Int64()
-
 	// if special values (latest) we return latest executed height
 	//
 	// all the special values are:
+	//	EarliestBlockNumber  = BlockNumber(-5)
 	//	SafeBlockNumber      = BlockNumber(-4)
 	//	FinalizedBlockNumber = BlockNumber(-3)
 	//	LatestBlockNumber    = BlockNumber(-2)
 	//	PendingBlockNumber   = BlockNumber(-1)
 	//
 	// EVM on Flow does not have these concepts, but the latest block is the closest fit
-	if height < 0 {
-		executed, err := blocksDB.LatestEVMHeight()
+	height := uint64(blockNumber)
+	var err error
+	if blockNumber == rpc.EarliestBlockNumber {
+		height = 0
+	} else if blockNumber <= rpc.PendingBlockNumber {
```

Comment:
```suggestion
	} else if blockNumber < 0 {
```

Optional, but I think this would be slightly safer.

Reply:
I have updated the implementation in this commit: https://github.com/onflow/flow-evm-gateway/pull/805/commits/c95ea08830118552f3fbb6f9e36b8344f66e1a54, to make it more explicit and readable. I want to avoid using magic values like `0`, I would prefer making comparisons with the constants defined by Geth.